### PR TITLE
Add Whisper Speech-to-Text

### PR DESCRIPTION
**Pull request by abheesht17**

### CODE REVIEW

File: keras_nlp/models/whisper/whisper_audio_to_speech_lm.py [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+)
+from keras_nlp.models.whisper.whisper_backbone import WhisperBackbone
+from keras_nlp.models.whisper.whisper_presets import backbone_presets
+from keras_nlp.utils.keras_utils import is_xla_compatible
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioToSpeechLM")
+class WhisperAudioToSpeechLM(GenerativeTask):
```
Comment (mattdangerw) [inline keras_nlp/models/whisper/whisper_audio_to_speech_lm.py:35]: I think we should still probably call this seq2seq for consistency, after all, we didn't call it `BartTextToTextLM`.

File: keras_nlp/models/whisper/whisper_audio_to_speech_lm.py [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+    warranties or conditions of any kind. The underlying model is provided by a
+    third party and subject to a separate license, available
+    [here](https://github.com/facebookresearch/fairseq/).
+
+    Args:
+        backbone: A `keras_nlp.models.WhisperBackbone` instance.
+        preprocessor: A `keras_nlp.models.WhisperAudioToSpeechLMPreprocessor` or `None`.
+            If `None`, this model will not apply preprocessing, and inputs
+            should be preprocessed before calling the model.
+    """
```
Comment (mattdangerw) [inline keras_nlp/models/whisper/whisper_audio_to_speech_lm.py:66]: Given that we have presets, we can probably go ahead and add docstrings. Also keep in mind that after the multi-backend changes, we will favor `np.array` over `tf.constant`.

File: keras_nlp/models/whisper/whisper_audio_to_speech_lm_preprocessor.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+)
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioToSpeechLMPreprocessor")
+class WhisperAudioToSpeechLMPreprocessor(WhisperPreprocessor):
+    """Whisper AudioToSpeech LM preprocessor.
+
+    This layer is used as preprocessor for seq2seq tasks using the Whisper model.
```
Comment (mattdangerw) [inline keras_nlp/models/whisper/whisper_audio_to_speech_lm_preprocessor.py:36]: maybe use this language instead https://github.com/keras-team/keras-nlp/blob/03668b87b6aae0e4d87cda041e07e2dc81736331/keras_nlp/models/gpt2/gpt2_causal_lm_preprocessor.py#L33-L43

a little more up to date

File: keras_nlp/models/whisper/whisper_tokenizer.py [hunk 4]
```diff
@@ -113,6 +113,8 @@ def __init__(
         self.translate_token_id = special_tokens[translate_token]
         self.transcribe_token_id = special_tokens[transcribe_token]
 
+        self.end_token_id = self.eos_token_id
```
Comment (mattdangerw) [inline keras_nlp/models/whisper/whisper_tokenizer.py:116]: This is kind of an awkward line. Can we just update the whole layer to use `end_token_id` and `start_token_id`? That will be more consistent with the rest of the library.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
I think there are still some correctness issue we are debugging. But left some general code comments in meantime.

**Comment by abheesht17** [comment]
Closing this, will open a new PR.

**Pull request closed without merging**